client/my-sites/controller.js

```diff
@@ -116,10 +116,11 @@ function renderEmptySites( context ) {
 	);
 }
 
-function renderNoVisibleSites( context ) {
+function renderNoVisibleSites( context, getState ) {
 	const EmptyContentComponent = require( 'components/empty-content' );
 	const currentUser = user.get();
-	const hiddenSites = currentUser.site_count - currentUser.visible_site_count;
+	const visibleSiteCount = getVisibleSites( getState() ).length;
+	const hiddenSites = currentUser.site_count - visibleSiteCount;
```

Comment:
Interesting, so the current_user.visible_site_count is incorrect? Might be worth investigating what's happening on the /me endpoint as a follow up.